nuxeo-core/nuxeo-core-cache/src/main/java/org/nuxeo/ecm/core/cache/CacheMetrics.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * (C) Copyright 2017 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.nuxeo.ecm.core.cache;
+
+import java.io.Serializable;
+
+import org.nuxeo.runtime.metrics.MetricsService;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.RatioGauge;
+import com.codahale.metrics.SharedMetricRegistries;
+
+public class CacheMetrics extends CacheWrapper {
+
+    protected final MetricRegistry registry;
+
+    protected final Counter read;
+
+    protected final Counter read_hit;
+
+    protected final Counter read_miss;
+
+    protected final RatioGauge read_hit_ratio;
+
+    protected final Counter write;
+
+    protected final Counter invalidation;
+
+    protected final Gauge<Long> size;
+
+    protected final String READ_HIT_NAME = nameOf("read-hit-counter");
```

Comment:
These fields are initialized before the constructor body, and because `nameOf` needs the cache name which is initialized from the constructor body I'm not sure how this will work.